### PR TITLE
Repair and Improve GetDiffRangeWithWhitespaceBehavior

### PR DESCRIPTION
**Pull request by 6543**
make sure we dont have a "pipe leak" & open gitRepo once!

### CODE REVIEW

File: services/gitdiff/gitdiff.go [hunk 1]
```diff
@@ -1205,29 +1205,18 @@ func readFileName(rd *strings.Reader) (string, bool) {
...
+	repoPath := gitRepo.Path
 
 	commit, err := gitRepo.GetCommit(afterCommitID)
 	if err != nil {
 		return nil, err
 	}
 
-	// FIXME: graceful: These commands should likely have a timeout
+	// FIXME: graceful: These commands should likely have a timeout <- YES!!!
 	ctx, cancel := context.WithCancel(git.DefaultContext)
```
Comment (6543) [inline services/gitdiff/gitdiff.go]: TODO: lookup how to pas timeout into context

Comment (jolheiser) [reply services/gitdiff/gitdiff.go]: https://pkg.go.dev/context#WithTimeout

Comment (6543) [reply services/gitdiff/gitdiff.go]: @jolheiser thanks & done

**Comment by 6543** [comment]
-> #16895
-> #16896

**Review by zeripath — APPROVED** [review]
Yes this is a better way of doing things

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
.

**Pull request merged**